sdk/communication/azure-communication-administration/src/main/java/com/azure/communication/administration/PhoneNumberClientBuilder.java

```diff
@@ -0,0 +1,264 @@
+import com.azure.communication.administration.implementation.PhoneNumberAdminClientImplBuilder;
+import com.azure.communication.common.CommunicationClientCredential;
+import com.azure.communication.common.HmacAuthenticationPolicy;
+import com.azure.core.annotation.ServiceClientBuilder;
+import com.azure.core.http.HttpClient;
+import com.azure.core.http.HttpPipeline;
+import com.azure.core.http.HttpPipelineBuilder;
+import com.azure.core.http.policy.CookiePolicy;
+import com.azure.core.http.policy.HttpLogDetailLevel;
+import com.azure.core.http.policy.HttpLogOptions;
+import com.azure.core.http.policy.HttpLoggingPolicy;
+import com.azure.core.http.policy.HttpPipelinePolicy;
+import com.azure.core.http.policy.RetryPolicy;
+import com.azure.core.http.policy.UserAgentPolicy;
+import com.azure.core.util.Configuration;
+import com.azure.core.util.CoreUtils;
+import com.azure.core.util.logging.ClientLogger;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+
+/**
+ * Builder for creating clients of Communication Service phone number configuration
+ */
+@ServiceClientBuilder(serviceClients = {PhoneNumberClient.class, PhoneNumberAsyncClient.class})
+public final class PhoneNumberClientBuilder {
+    private static final Map<String, String> PROPERTIES =
+        CoreUtils.getProperties("azure-communication-administration.properties");
+    private static final String SDK_NAME = "name";
+    private static final String SDK_VERSION = "version";
+
+    private final ClientLogger logger = new ClientLogger(PhoneNumberClientBuilder.class);
+
+    private PhoneNumberServiceVersion version;
+    private String endpoint;
+    private HttpPipeline pipeline;
+    private HttpClient httpClient;
+    private HttpLogOptions httpLogOptions;
+    private CommunicationClientCredential credential;
+    private Configuration configuration;
+    private final List<HttpPipelinePolicy> additionalPolicies = new ArrayList<>();
+
+    /**
+     * Set endpoint of the service
+     *
+     * @param endpoint url of the service
+     * @return The updated {@link PhoneNumberClientBuilder} object.
+     * @throws NullPointerException If {@code endpoint} is {@code null}.
+     */
+    public PhoneNumberClientBuilder endpoint(String endpoint) {
+        this.endpoint = Objects.requireNonNull(endpoint, "'endpoint' cannot be null.");
+        return this;
+    }
+
+    /**
+     * Sets the HTTP pipeline to use for the service client
+     * <p>
+     * If {@code pipeline} is set, all other settings aside from
+     * {@link PhoneNumberClientBuilder#endpoint(String) endpoint} are ignored.
+     *
+     * @param pipeline HttpPipeline to use
+     * @return The updated {@link PhoneNumberClientBuilder} object.
+     */
+    public PhoneNumberClientBuilder pipeline(HttpPipeline pipeline) {
+        this.pipeline = Objects.requireNonNull(pipeline, "'pipeline' cannot be null.");
```

Comment:
This can be null in scenarios where the user initially set the pipeline but then decided to clear it out. These validations can be postponed until the build method is called.

Reply:
To be addressed in separate PR.